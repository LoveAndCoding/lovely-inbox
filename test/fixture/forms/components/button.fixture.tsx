import * as React from "react";
import * as TestRenderer from "react-test-renderer";

import LovelyButton, {
	ButtonColors,
	ButtonSizes,
	ButtonTypes,
} from "src/ui/forms/components/button";

describe("<LovelyButton>", () => {
	test("Default props render correctly", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<LovelyButton>Default Button</LovelyButton>,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Color rendering changed via prop", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<LovelyButton color={ButtonColors.light}>
				Light Button
			</LovelyButton>,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Size rendering changed via prop", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<LovelyButton size={ButtonSizes.xsmall}>
				XSmall Button
			</LovelyButton>,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Type rendering changed via prop", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<LovelyButton type={ButtonTypes.submit}>
				Submit Button
			</LovelyButton>,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Disabled rendering changed via prop", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<LovelyButton disabled={true}>Submit Button</LovelyButton>,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});
});
