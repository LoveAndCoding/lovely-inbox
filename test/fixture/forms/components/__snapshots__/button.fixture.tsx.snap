// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LovelyButton> Color rendering changed via prop 1`] = `
<button
  className="button_1bs4i0o-o_O-medium_e28yzh-o_O-light_13slhe1"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Light Button
</button>
`;

exports[`<LovelyButton> Default props render correctly 1`] = `
<button
  className="button_1bs4i0o-o_O-medium_e28yzh-o_O-dark_xzwfai"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Default Button
</button>
`;

exports[`<LovelyButton> Disabled rendering changed via prop 1`] = `
<button
  className="button_1bs4i0o-o_O-medium_e28yzh-o_O-dark_xzwfai-o_O-disabled_1lrpyjs"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  Submit Button
</button>
`;

exports[`<LovelyButton> Size rendering changed via prop 1`] = `
<button
  className="button_1bs4i0o-o_O-xsmall_ba50ph-o_O-dark_xzwfai"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  XSmall Button
</button>
`;

exports[`<LovelyButton> Type rendering changed via prop 1`] = `
<button
  className="button_1bs4i0o-o_O-medium_e28yzh-o_O-dark_xzwfai"
  disabled={false}
  onClick={[Function]}
  type="submit"
>
  Submit Button
</button>
`;
